### PR TITLE
fix: LanceDB deleteByFile case-sensitivity + core test suite

### LINKED ISSUE #121: Bug: LanceDB `deleteByFile` query fails due to DataFusion case-sensitivity/parsing edge cases
## Context
During incremental `totem sync`, the ingestion pipeline calls `deleteByFile` to remove stale chunks before re-embedding updated files. LanceDB relies on DataFusion for its SQL querying backend. DataFusion has strict, idiosyncratic case-sensitivity rules for column names (it often requires explicit quoting to preserve camelCase like `filePath`). 

If this query fails due to parsing errors or case-mismatches, incremental syncs will fail to delete old data. Over time, the local vector index will drift from the actual codebase, causing the orchestrator to ingest poisoned, duplicate, or outdated context.

## Acceptance Criteria
- [ ] Investigate the `deleteByFile` query syntax in `packages/core/src/store/lance-store.ts`.
- [ ] Ensure the `filePath` column name (and the file path string value itself) is properly escaped/quoted to satisfy DataFusion's parser.
- [ ] Write a test that explicitly verifies `deleteByFile` successfully removes records from a live LanceDB instance.
- [ ] Verify that paths with spaces, capital letters, and backslashes (on Windows) are handled correctly during deletion.

### LINKED ISSUE #122: Test: Backfill unit and integration tests for `@mmnto/totem` core package
## Context
The `@mmnto/cli` package has robust unit testing (118+ tests), but the `@mmnto/totem` core package currently has zero test coverage.

This package contains the most complex and critical logic in the system:
- Syntactic Chunkers (AST, Markdown headings, Session logs)
- The LanceDB Database Interface
- The Orchestrator Embedding Pipeline (OpenAI / Ollama)

Without tests here, regressions in AST parsing or vector storage state are incredibly difficult to catch before they reach the user.

## Acceptance Criteria
- [ ] **Setup:** Configure Vitest infrastructure in `packages/core/`.
- [ ] **Unit Tests (Chunkers):** Add tests for `markdown-chunker.ts` and `typescript-chunker.ts` to ensure they correctly split files based on semantic boundaries rather than blind character limits.
- [ ] **Integration Test (LanceDB):** Create a basic E2E integration test that runs the `sync` pipeline against a temporary, in-memory/temp-folder LanceDB instance, verifying that chunks are correctly embedded and retrievable via `search_knowledge`.

### PR DESCRIPTION
**Pull request by satur8d**
## Summary

- **Bug fix (#121):** `deleteByFile` used unquoted `filePath` in the DataFusion SQL filter, which silently resolved to lowercase `filepath` and matched nothing. Fixed by using backtick-quoted identifier (`` `filePath` ``). Double-quote quoting was also tested and found to silently fail — DataFusion uses backticks, not SQL-standard double quotes.
- **Test suite (#122):** Added vitest infrastructure and 36 tests for the `@mmnto/totem` core package, which previously had zero test coverage.
- **Docs:** Phase 4 Federated Memory vision added to architecture.md and roadmap.md.

Closes #121
Closes #122

## Changes

| Area | What |
|------|------|
| `lance-store.ts:93` | `` `filePath` `` backtick quoting in `deleteByFile` filter |
| `lance-store.test.ts` | 15 integration tests (insert, delete, search, stats, reset, reconnect) |
| `markdown-chunker.test.ts` | 9 unit tests (heading splits, frontmatter, line numbers, edge cases) |
| `typescript-chunker.test.ts` | 12 unit tests (functions, classes, interfaces, hooks, components) |
| `vitest.config.ts` | Vitest setup for core package |
| `package.json` | Added vitest dev dep, updated test script |

## Key Finding

LanceDB/DataFusion identifier quoting uses **backticks** (`` ` ``), not SQL-standard double quotes (`"`). Double-quoted identifiers silently produce zero matches without throwing errors — a particularly nasty silent failure mode.

## Test plan

- [x] All 36 new core tests pass
- [x] All 118 existing CLI tests pass (154 total)
- [x] `pnpm build` succeeds across all packages
- [x] Pre-push hooks pass (format, lint, test)
- [x] `totem shield` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/core/src/chunkers/typescript-chunker.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    expect(chunks.length).toBe(1);
+    expect(chunks[0]!.label).toBe('enum: Direction');
+  });
+
+  it('detects React components (PascalCase arrow functions)', () => {
+    const code = `export const Dashboard = () => {
+  return <div>Dashboard</div>;
+};
+`;
+    const chunks = chunker.chunk(code, 'src/Dashboard.tsx', 'code');
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/core/src/chunkers/typescript-chunker.test.ts]: @gemini-code-assist Declining — the styleguide says *this project* does not use .tsx, but the TypeScriptChunker is designed to process *consumer* codebases that do. This test validates the ScriptKind.TSX branch (typescript-chunker.ts:18). Removing .tsx coverage would leave that code path untested.

File: packages/core/src/store/lance-store.test.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+      await store.deleteByFile("src/user's-config.ts");
+
+      const stats = await store.stats();
+      expect(stats.totalChunks).toBe(1);
+    });
+
+    it('handles camelCase paths', async () => {
+      await store.insert([
+        makeChunk({ filePath: 'src/myComponent/CamelCase.tsx', content: 'camel' }),
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/core/src/store/lance-store.test.ts]: @gemini-code-assist Same rationale as the tsx comment above — this tests deleteByFile with a path a consumer project might have. The extension is irrelevant to the delete logic being tested; the camelCase directory name is what matters.
